File: src/main/java/bad/robot/radiate/teamcity/RunInformation.java
package bad.robot.radiate.teamcity;

public class RunInformation extends TeamCityObject {

    private final Integer percentageComplete;
    private final Integer elapsedSeconds;
    private final Integer estimatedTotalSeconds;
    private final Boolean outdated;
    private final Boolean probablyHanging;

    public RunInformation(Integer percentageComplete, Integer elapsedSeconds, Integer estimatedTotalSeconds, Boolean outdated, Boolean probablyHanging) {
        this.percentageComplete = percentageComplete;
        this.elapsedSeconds = elapsedSeconds;
        this.estimatedTotalSeconds = estimatedTotalSeconds;
        this.outdated = outdated;
        this.probablyHanging = probablyHanging;
    }

    public Integer getPercentageComplete() {
        return percentageComplete;
    }

    Integer getElapsedSeconds() {
        return elapsedSeconds;
    }

    Integer getEstimatedTotalSeconds() {
        return estimatedTotalSeconds;
    }

    public Boolean getOutdated() {
        return outdated;
    }

    public Boolean getProbablyHanging() {
        return probablyHanging;
    }
}

File: src/main/scala/bad/robot/radiate/teamcity/RunInformation.scala
package bad.robot.radiate.teamcity

import argonaut.DecodeJson

object RunInformation {
  implicit def runInformationDecoder: DecodeJson[RunInformation] = {
    DecodeJson(cursor => {
      for {
        percentageComplete <- cursor.get[Int]("percentageComplete")
        elapsedSeconds <- cursor.get[Int]("elapsedSeconds")
        estimatedTotalSeconds <- cursor.get[Int]("estimatedTotalSeconds")
        outdated <- cursor.get[Boolean]("outdated")
        probablyHanging <- cursor.get[Boolean]("probablyHanging")
      } yield {
        RunInformation(percentageComplete, elapsedSeconds, estimatedTotalSeconds, outdated, probablyHanging)
      }
    })
  }
}

case class RunInformation(percentageComplete: Integer, elapsedSeconds: Integer, estimatedTotalSeconds: Integer, outdated: Boolean, probablyHanging: Boolean) extends TeamCityObject
